### PR TITLE
review and simplify the Platform Creation page

### PR DESCRIPTION
**Pull request by frivieremicroej**
- Simplify required steps [create an empty project + PCA download +
build]. No more Architecture manual download + import + "New Platform
Project" wizard
- explain Platform module and then manual import for Platform Editor
- various (as follows...)

### CODE REVIEW

File: PlatformDeveloperGuide/platformCreation.rst [hunk 1]
```diff
@@ -144,94 +141,78 @@ The next step is to create a MicroEJ Platform configuration:
 
      </dependencies>
 
-  For example, to declare the MicroEJ Architecture Specific Pack UI
-  version ``13.0.4`` for MicroEJ Architecture ``flopi4G25`` on Arm®
-  Cortex®-M4 microcontrollers compiled with GNU CC toolchain:
+- :ref:`Build the Platform <platform_build>` again.
```
Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:144]: This line should be moved after the packs examples.

Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:144]: Shouldn't we indicate to refresh the platform project here ?

File: PlatformDeveloperGuide/platformCreation.rst [hunk 2]
```diff
@@ -144,94 +141,78 @@ The next step is to create a MicroEJ Platform configuration:
...
 
-MicroEJ Platform Build
-======================
+.. _platform_module_configuration:
 
+Platform Module Configuration
+=============================
 
-To build the MicroEJ Platform, perform as a regular :ref:`mmm_module_build`: 
+A Platform module is the minimal unit that can extend a MicroEJ Architecture with additional features such as:
```
Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:186]: The link between a Platform module, a Pack and a Architecture is not easy to make. The introduction sentence of the previous chapter states that a pack "provides additional features on top of the MicroEJ Architecture", and here we say that a Platform module also add features to the kernel. We should at least add a sentence to explain that packs are made of Platform modules to make the link more clear.

File: PlatformDeveloperGuide/platformCreation.rst [hunk 3]
```diff
@@ -144,94 +141,78 @@ The next step is to create a MicroEJ Platform configuration:
...
-        [echo]     To import this project in your MicroEJ SDK workspace (if not already available):
-        [echo]      - Select 'File' > 'Import...' > 'General' > 'Existing Projects into Workspace' > 'Next'
-        [echo]      - Check 'Select root directory' and browse 'C:\tmp\mydevice-Platform-mytoolchain-0.0.1' > 'Finish'
-        [echo]     ============================================================================================================
+Platform modules provided by :ref:`MicroEJ Generic Packs <pack_generic>` are automatically installed during the :ref:`Platform build <platform_build>` 
+and do not require extra configuration. They are not displayed in the Platform Editor.
 
-      BUILD SUCCESSFUL
+Platform modules provided by :ref:`MicroEJ Architectures <architecture_overview>`, :ref:`MicroEJ Architecture Specific Packs <pack_architecture_specific>`
+and :ref:`Legacy MicroEJ Generic Packs <pack_generic_legacy>` following list are **not installed** by default.
```
Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:197]: Is "following list" added by error ?

File: PlatformDeveloperGuide/platformCreation.rst [hunk 4]
```diff
@@ -144,94 +141,78 @@ The next step is to create a MicroEJ Platform configuration:
...
-   MicroEJ Platform Project
- 
-This step is only required the first time the Platform is built, or if the Platform properties haved changed. 
-When the same Platform is rebuilt, right-click on the Platform project and select :guilabel:`Refresh` to get the new content.
+Before opening the Platform Editor, the Platform must have been built once to let :ref:`mmm` resolve and download MicroEJ Architecture and Packs locally.
+Then import them in MicroEJ SDK as follows:
 
-.. _platform_configuration_modules:
+- Select :guilabel:`File` > :guilabel:`Import` > :guilabel:`MicroEJ` > :guilabel:`Architectures`,
+- Browse :guilabel:`myplatform-configuration\target~\dependencies` folder (contains ``.xpf`` and ``.xpfp`` files once the Platform is built),
```
Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:204]: `myplatform` is never mentioned before. We should add something like "let's call it `myplatform`" at the beginning of the tutorial when creating the project.

Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:204]: `\` are not displayed in the html version, they should be doubled, or even better, replaced by `/`.

File: PlatformDeveloperGuide/platformCreation.rst [hunk 5]
```diff
@@ -144,94 +141,78 @@ The next step is to create a MicroEJ Platform configuration:
...
+- Check the :guilabel:`I agree and accept the above terms and conditions...` box to accept the license,
+- Click on :guilabel:`Finish` button. This may take some time.
 
-Platform Groups / Modules Selection
-===================================
+Once imported, double-click on the :guilabel:`[name].platform` file to open the Platform Editor.
```
Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:208]: I have a `default.platform` file, so not the name of the platform, is it expected ?

File: PlatformDeveloperGuide/platformCreation.rst [hunk 6]
```diff
@@ -240,62 +221,56 @@ imported.
...
+folder named after the module beside the :guilabel:`[name].platform`
 definition. It may contain:
 
--  An optional [module].properties file named after the module name.
+-  A :guilabel:`[module].properties` file named after the module name.
    These properties will be injected in the execution context prefixed
    by the module name. Some properties might be needed for the
    configuration of some modules. Please refer to the modules
    documentation for more information.
+-  A :guilabel:`bsp.xml` file which provide additional information about the BSP
```
Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:241]:
```suggestion
-  A :guilabel:`bsp.xml` file which provides additional information about the BSP
```

File: PlatformDeveloperGuide/platformCreation.rst [hunk 7]
```diff
@@ -240,62 +221,56 @@ imported.
...
-   </project>
+  It is possible to quickly rebuild the Platform from the Platform Editor if only Platform module configuration has changed.
+  Click on the :guilabel:`Build Platform` link on the Platform configuration :guilabel:`Overview` tab.
+
+.. _platformCustomization:
+
+Platform Customization
+======================
 
 Configuration project (the project which contains the
```
Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:275]:
```suggestion
The configuration project (the project which contains the
```

File: PlatformDeveloperGuide/platformCreation.rst [hunk 8]
```diff
@@ -240,62 +221,56 @@ imported.
...
-   
-   	<!--
-   		Define "project.dir" property that references the directory 
-   		where this file is located.
-   	-->
-   	<dirname property="project.dir" file="${ant.file.configuration}"/>
-   
-   </project>
+  It is possible to quickly rebuild the Platform from the Platform Editor if only Platform module configuration has changed.
+  Click on the :guilabel:`Build Platform` link on the Platform configuration :guilabel:`Overview` tab.
```
Comment (tdelhomenie-microej) [inline PlatformDeveloperGuide/platformCreation.rst:268]: It is not clear what is the purpose of the checkboxes of the modules. They are not included if not checked ?

**Review by tdelhomenie-microej — COMMENTED** [review]
Other comments:
- This chapter structure is weird, it starts with a description, then it lists steps:
```
The dropins folder organization should respect the final Platform files and folders organization. For instance, the tools are located in the sub-folder tools. Launch a Platform build without the dropins folder to see how the Platform files and folders organization is. Then fill the dropins folder with additional features and build again the Platform to obtain an advanced Platform.
```

File: PlatformDeveloperGuide/platformCreation.rst [hunk 9]
```diff
@@ -12,76 +12,31 @@ and options to connect it to an external Board Support Package (BSP) as well as
...
-   mandatory, you should therefore set them. Note that their values can
-   be modified later on.
+-  Enter a :guilabel:`Project name`. The name is arbitrary and can be changed later. The usual convention is ``[PLATFORM_NAME]-configuration``,
 
--  Click on :guilabel:`Finish` button. A new project ``[device]-[name]-[toolchain]`` is being created
-   containing a ``[name].platform`` file. The Platform Editor shall then
-   open.
+-  Click on :guilabel:`Finish` button. A new empty project is created,
 
 -  Install `Platform Configuration Additions <https://github.com/MicroEJ/PlatformQualificationTools/blob/master/framework/platform/>`_.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/platformCreation.rst:41]: Here we want to install the latest available version (the PCA version is stored in https://github.com/MicroEJ/PlatformQualificationTools/blob/master/framework/platform/content/build/CHANGELOG.md)

File: PlatformDeveloperGuide/platformCreation.rst (hunk 5 above)
Comment (frivieremicroej) [reply PlatformDeveloperGuide/platformCreation.rst]: renamed

File: PlatformDeveloperGuide/platformCreation.rst (hunk 1 above)
Comment (frivieremicroej) [reply PlatformDeveloperGuide/platformCreation.rst]: done

File: PlatformDeveloperGuide/platformCreation.rst (hunk 8 above)
Comment (frivieremicroej) [reply PlatformDeveloperGuide/platformCreation.rst:266]: it is explained above, maybe not enough clear, TBD

File: ApplicationDeveloperGuide/limitations.rst [hunk 10]
```diff
@@ -4,7 +4,7 @@ Limitations
 ===========
 
 The following table lists the limitations of MicroEJ Architectures version ``7.14.0`` or higher, for both Evaluation and Production usage.
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/limitations.rst:6]: Remains to reintegrate / merge wording fixes of https://github.com/MicroEJ/docs/pull/277

File: PlatformDeveloperGuide/platformCreation.rst [hunk 11]
```diff
@@ -127,10 +76,58 @@ The next step is to create a MicroEJ Platform configuration:
...
-- Edit the :ref:`mmm_module_description` ``module.ivy`` to declare the dependency line to the MicroEJ Packs previously downloaded:
+.. _platform_build:
+
+Platform Build
+==============
 
+To build the MicroEJ Platform, perform as a regular :ref:`mmm_module_build`: 
+
+  - Right-click on the Platform Configuration project,
+  - Select :guilabel:`Build Module`.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/platformCreation.rst:87]: OK I will invert pack import and platform build

File: PlatformDeveloperGuide/platformCreation.rst (hunk 1 above)
Comment (frivieremicroej) [reply PlatformDeveloperGuide/platformCreation.rst]: Once imported, the Eclipse project should automatically refresh
OK for testing again

**Comment by frivieremicroej** [comment]
Doc Build Link https://docs.microej.com/en/feature-m0078ddg-409_move_manual_download/

File: PlatformDeveloperGuide/platformCreation.rst [hunk 12]
```diff
@@ -95,13 +50,7 @@ The next step is to create a MicroEJ Platform configuration:
...
       MicroEJ Platform Configuration Project Skeleton
 
-- Edit the ``module.properties`` file and set the option ``com.microej.platformbuilder.platform.filename`` to the ``[name].platform`` file name.
-
-  .. code-block::
-
-     com.microej.platformbuilder.platform.filename=myplatform.platform
-
-- Edit the :ref:`mmm_module_description` ``module.ivy`` to declare the dependency line to the MicroEJ Architecture previously downloaded:
+- Edit the :ref:`mmm_module_description` ``module.ivy`` to declare the MicroEJ Architecture dependency:
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply PlatformDeveloperGuide/platformCreation.rst:57]: Same comment for example of pack dependency that need to be removed.
=> https://youtrack.cross/issue/M0018PQT-319

**Pull request merged**